source/AMDSClient.cpp

```diff
@@ -0,0 +1,347 @@
+			this, SLOT(enableGetFortuneButton()));
+	connect(portLineEdit, SIGNAL(textChanged(QString)),
+			this, SLOT(enableGetFortuneButton()));
+	connect(getFortuneButton, SIGNAL(clicked()),
+			this, SLOT(requestNewFortune()));
+	connect(quitButton, SIGNAL(clicked()), this, SLOT(close()));
+	connect(tcpSocket, SIGNAL(readyRead()), this, SLOT(readFortune()));
+	connect(tcpSocket, SIGNAL(error(QAbstractSocket::SocketError)),
+			this, SLOT(displayError(QAbstractSocket::SocketError)));
+
+	QGridLayout *mainLayout = new QGridLayout;
+	mainLayout->addWidget(hostLabel, 0, 0);
+	mainLayout->addWidget(hostLineEdit, 0, 1);
+	mainLayout->addWidget(portLabel, 1, 0);
+	mainLayout->addWidget(portLineEdit, 1, 1);
+	mainLayout->addWidget(statusLabel, 2, 0, 1, 2);
+	mainLayout->addWidget(buttonBox, 3, 0, 1, 2);
+
+	QFormLayout *formLayout = new QFormLayout();
+	formLayout->addRow("Request Type", requestType);
+	formLayout->addRow("Buffer", bufferNameComboBox_);
+	formLayout->addRow("Time1", time1Edit);
+	formLayout->addRow("Time2", time2Edit);
+	formLayout->addRow("Count1", count1Edit);
+	formLayout->addRow("Count2", count2Edit);
+	formLayout->addRow("Include Status", includeStatusDataCheckbox);
+	mainLayout->addLayout(formLayout, 4, 0, 1, 2);
+	mainLayout->addWidget(results, 5, 0, 1, 2);
+	setLayout(mainLayout);
+
+	setWindowTitle(tr("Fortune Client"));
+	portLineEdit->setFocus();
+
+	QNetworkConfigurationManager manager;
+	if (manager.capabilities() & QNetworkConfigurationManager::NetworkSessionRequired) {
+		// Get saved network configuration
+		QSettings settings(QSettings::UserScope, QLatin1String("Trolltech"));
+		settings.beginGroup(QLatin1String("QtNetwork"));
+		const QString id = settings.value(QLatin1String("DefaultNetworkConfiguration")).toString();
+		settings.endGroup();
+
+		// If the saved network configuration is not currently discovered use the system default
+		QNetworkConfiguration config = manager.configurationFromIdentifier(id);
+		if ((config.state() & QNetworkConfiguration::Discovered) !=
+			QNetworkConfiguration::Discovered) {
+			config = manager.defaultConfiguration();
+		}
+
+		networkSession = new QNetworkSession(config, this);
+		connect(networkSession, SIGNAL(opened()), this, SLOT(sessionOpened()));
+
+		getFortuneButton->setEnabled(false);
+		statusLabel->setText(tr("Opening network session."));
+		networkSession->open();
+	}
+}
+
+void AMDSClient::requestNewFortune()
+{
+	getFortuneButton->setEnabled(false);
+	blockSize = 0;
+	tcpSocket->abort();
+	tcpSocket->connectToHost(hostLineEdit->text(),
+							 portLineEdit->text().toInt());
+
+	QTimer::singleShot(1000, this, SLOT(requestData()));
+}
+
+void AMDSClient::readFortune()
+{
+
+	AMDSDataStream in(tcpSocket);
+	in.setVersion(QDataStream::Qt_4_0);
+
+	if (blockSize == 0) {
+		if (tcpSocket->bytesAvailable() < (int)sizeof(quint32))
+			return;
+
+		in >> blockSize;
+	}
+
+//	qDebug() << "Received from: " << tcpSocket->peerAddress();
```

Comment:
There are some debugs here.  If you don't need them anymore then you could remove them.  Otherwise they are fine as they are.